narayan954/dummygram
# feat:  initial commitlint configuration ✨

**Related Issues:**
Closes #898 

### **Description:**
This pull request introduces the integration of Commitlint into the project. Commitlint is a powerful tool that enforces a standardized format for commit messages, ensuring consistency and clarity across all contributions. By adhering to a common commit message convention, we can enhance collaboration, simplify code review processes, and facilitate better changelog generation.

### **Changes:**
- Added the necessary Commitlint configuration files.
- Set up commit message rules to enforce the conventional commit format.
- Updated the project's contributing guidelines to include Commitlint usage.

**Testing:**
To ensure that Commitlint is functioning correctly, we have conducted extensive testing on this implementation. We have created test commits with various scenarios, including valid and invalid commit messages, to verify that the commit message rules are being enforced accurately. All tests have successfully passed, demonstrating the effectiveness of Commitlint in maintaining commit message consistency.

### **How to Verify:**
1. Clone this branch and run the project locally.
2. Make a test commit with a valid commit message format.
3. Make another test commit with an invalid commit message format (non-compliant with the conventional commit format).
4. Observe the output of Commitlint during the commit process and verify that it accepts the valid message and rejects the invalid one.

Please review this pull request, and let me know if any further adjustments or improvements are required. Your feedback is highly appreciated, and together, we can ensure a more organized and collaborative development process. Thank you!
